src/java/org/apache/cassandra/repair/RepairRunnable.java

```diff
@@ -141,11 +140,14 @@ protected void fireErrorAndComplete(int progressCount, int totalProgress, String
     static class CommonRange
     {
         public final Set<InetAddressAndPort> endpoints;
+        public final Set<InetAddressAndPort> transEndpoints;
         public final Collection<Range<Token>> ranges;
 
-        public CommonRange(Set<InetAddressAndPort> endpoints, Collection<Range<Token>> ranges)
+        public CommonRange(Set<InetAddressAndPort> endpoints, Set<InetAddressAndPort> transEndpoints, Collection<Range<Token>> ranges)
```

Comment:
Common range could use a comment basically saying that it aggregates the set of full and transiently replicating endpoints that match for a set of ranges. So it's a one to many relationship between two non-overlapping sets of endpoints (transient and full), this is the one, to the ranges they all replicate.

The relationship would be even clearer if it were a key value relationship instead of a collection/list.